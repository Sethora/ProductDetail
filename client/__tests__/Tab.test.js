import React from 'react';
import '@testing-library/jest-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Tab from '../src/components/tab';

afterEach(cleanup);

describe('Tab component', () => {
  it('renders', () => {
    const { asFragment } = render(<Tab
      value={'Detail'}
      position={0}
      selected={0}
      click={() => console.log('clicked')}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('gets the title from props', () => {
    const { getByRole } = render(<Tab
      value={'Detail'}
      position={0}
      selected={0}
      click={() => console.log('clicked')}
    />);
    expect(getByRole('title')).toHaveTextContent('Detail');
  });

  it('should have a click event', () => {
    let count = 0;
    const { getByTestId } = render(<Tab
      value={'Detail'}
      position={0}
      selected={0}
      click={() => count++}
    />);
    const TabStyles = getByTestId('tabstyles');
    fireEvent.click(TabStyles);
    expect(count).toBe(1);
  });
});