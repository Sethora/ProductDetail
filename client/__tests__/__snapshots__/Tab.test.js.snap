// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hrmhkb"
    data-testid="tabstyles"
  >
    <div
      class="clicked"
      role="title"
    >
      Detail
    </div>
    <div
      class="divider"
    />
  </div>
</DocumentFragment>
`;
